src/gui/http.go

```diff
@@ -27,44 +27,65 @@ var (
 	quit     chan struct{}
 )
 
+type Server struct {
+	daemon   *daemon.Daemon
+	listener net.Listener
+	quit     chan struct{}
+	appLoc   string
+	Err      error
+}
+
 const (
 	resourceDir = "dist/"
 	devDir      = "dev/"
 	indexPage   = "index.html"
 )
 
+func Create(needHttps bool, host string, guiDirectory string, daemon *daemon.Daemon, cert string, key string) *Server {
+	s := Server{
+		daemon: daemon,
+		quit:   make(chan struct{}),
+	}
+	if needHttps {
+		s.Err = s.launchWebInterfaceHTTPS(host, guiDirectory, daemon, cert, key)
+	} else {
+		s.Err = s.launchWebInterface(host, guiDirectory, daemon)
+	}
+	return &s
+}
+
 // LaunchWebInterface begins listening on http://$host, for enabling remote web access
 // Does NOT use HTTPS
-func LaunchWebInterface(host, staticDir string, daemon *daemon.Daemon) error {
-	quit = make(chan struct{})
+func (s *Server) launchWebInterface(host, staticDir string, daemon *daemon.Daemon) error {
+	var err error
 	logger.Info("Starting web interface on http://%s", host)
 	logger.Warning("HTTPS not in use!")
-	appLoc, err := file.DetermineResourcePath(staticDir, resourceDir, devDir)
+	s.appLoc, err = file.DetermineResourcePath(staticDir, resourceDir, devDir)
 	if err != nil {
 		return err
 	}
-	logger.Info("Web resources directory: %s", appLoc)
+	logger.Info("Web resources directory: %s", s.appLoc)
 
-	listener, err = net.Listen("tcp", host)
+	s.listener, err = net.Listen("tcp", host)
 	if err != nil {
 		return err
 	}
 
 	// Runs http.Serve() in a goroutine
-	serve(listener, NewGUIMux(appLoc, daemon), quit)
+	//s.serve(NewGUIMux(appLoc, daemon, &s))
 	return nil
 }
 
 // LaunchWebInterfaceHTTPS begins listening on https://$host, for enabling remote web access
 // Uses HTTPS
-func LaunchWebInterfaceHTTPS(host, staticDir string, daemon *daemon.Daemon, certFile, keyFile string) error {
-	quit = make(chan struct{})
+func (s *Server) launchWebInterfaceHTTPS(host, staticDir string, daemon *daemon.Daemon, certFile, keyFile string) error {
+	var err error
 	logger.Info("Starting web interface on https://%s", host)
```

Comment:
Move this log line to "Serve"
Add new log line that says "Web interface tcp listening on %s host"

Reply:
Do the same for `launchWebInterfaceHTTP`